src/Microsoft.DotNet.Darc/tests/Microsoft.DotNet.Darc.VirtualMonoRepo.E2E.Tests/VmrTests.cs

```diff
@@ -0,0 +1,528 @@
+        {
+            expectedFiles.Add(vmrPath / "src" / repo / "eng" / "Version.Details.xml");
+            expectedFiles.Add(vmrPath / "git-info" / $"{repo}.props");
+        }
+        
+        expectedFiles.AddRange(reposFiles);
+        
+        return expectedFiles;
+    }
+
+    private void CheckDirectoryContents(string directory, IList<LocalPath> expectedFiles)
+    {
+        var filesInDir = GetAllFilesInDirectory(new DirectoryInfo(directory));
+        filesInDir.Should().BeEquivalentTo(expectedFiles);
+    }
+
+    private static void CheckFileContents(LocalPath filePath, string expected)
+    {
+        var fileContent = File.ReadAllText(filePath);
+        fileContent.Should().BeEquivalentTo(expected);
+    }
+
+    private async Task CheckAllIsCommited(string repo)
+    {
+        var gitStatus = await _processManager.ExecuteGit(repo, "status", "--porcelain");
+        gitStatus.StandardOutput.Should().Be(string.Empty);
+    }
+
+    private async Task CallDarcInitialize(string vmrPath, string tmpPath, string repository, string commit)
+    {
+        await CallDarcVmrCommand("initialize", vmrPath, tmpPath, new[] { $"{repository}:{commit}" });
+    }
+
+    private async Task CallDarcUpdate(string vmrPath, string tmpPath, string repository, string commit)
+    {
+        await CallDarcVmrCommand("update", vmrPath, tmpPath, new[] { $"{repository}:{commit}" });
+    }
+
+    private async Task CallDarcVmrCommand(string command, string vmrPath, string tmpPath, string[] arguments)
+    {
+        var args = new List<string>
+        {
+            _darcDll,
+            "vmr",
+            command,
+            "--recursive",
+            "--vmr",
+            vmrPath,
+            "--tmp",
+            tmpPath
+        };
+
+        args.AddRange(arguments);
+
+        var res = await _processManager.Execute("dotnet", args);
+        //res.ExitCode.Should().Be(0, res.StandardError);
+        Assert.True(res.ExitCode == 0, res.StandardError);
+    }
+
+    private async Task ConfigureGit(LocalPath repo)
+    {
+        await _processManager.ExecuteGit(repo, "config", "user.email", Constants.DarcBotEmail);
+        await _processManager.ExecuteGit(repo, "config", "user.name", Constants.DarcBotName);
+    }
+
+    private async Task CommitAll(LocalPath repo, string commitMessage)
+    {
+        await _processManager.ExecuteGit(repo, "add", "-A");
+        await _processManager.ExecuteGit(repo, "commit", "-m", commitMessage);
+    }
+
+    private async Task InitialCommit(LocalPath repo)
+    {
+        await _processManager.ExecuteGit(repo, "init", "-b", "main");
+        await ConfigureGit(repo);
+        await CommitAll(repo, "Initial commit");
+    }
+
+    private async Task<string> GetRepoLastCommit(LocalPath repo)
+    {
+        var log = await _processManager.ExecuteGit(repo, "log", "--format=format:%H");
+        return log.StandardOutput.Split("\r\n", StringSplitOptions.RemoveEmptyEntries).First();
```

Comment:
Let's make this work for Linux too:

```suggestion
        return log.StandardOutput.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries).First();
```